templates/vue-starter-template/app/components/form/BaseInput.vue

```diff
@@ -1,6 +1,8 @@
 <script lang="ts" setup>
-defineProps<{
+const { placeholder, type = "text" } = defineProps<{
   placeholder?: string;
+  type?: "text" | "password";
```

Comment:
can't we be more open to all types of inputs?
maybe something like HTMLInputElement['type'] or something similar for autocompletion?

Reply:
We cannot use "global type" because this component should not handle all types, like the `color`, `calendar`. I added additional types.